examples/gno.land/r/demo/hof/hof_test.gno

```diff
@@ -0,0 +1,135 @@
+package hof
+
+import (
+	"std"
+	"testing"
+
+	"gno.land/p/demo/testutils"
+	"gno.land/p/demo/uassert"
+	"gno.land/p/demo/urequire"
+)
+
+const rlmPath = "gno.land/r/gnoland/home"
+
+var (
+	admin      = owner.Owner()
+	adminRealm = std.NewUserRealm(admin)
+	alice      = testutils.TestAddress("alice")
+)
+
+func TestRegister(t *testing.T) {
+
+	// Test user realm register
+	aliceRealm := std.NewUserRealm(alice)
+	std.TestSetRealm(aliceRealm)
+
+	Register()
+	uassert.False(t, itemExists(t, rlmPath))
+
+	// Test register while paused
+	std.TestSetRealm(adminRealm)
+	Pause()
+
+	// Set legitimate caller
+	std.TestSetRealm(std.NewCodeRealm(rlmPath))
+
+	Register()
+	uassert.False(t, itemExists(t, rlmPath))
+
+	// Unpause
+	std.TestSetRealm(adminRealm)
+	Unpause()
+
+	// Set legitimate caller
+	std.TestSetRealm(std.NewCodeRealm(rlmPath))
+	Register()
+
+	// Find registered items
+	uassert.True(t, itemExists(t, rlmPath))
+}
+
+func TestUpvote(t *testing.T) {
+	raw, _ := exhibition.items.Get(rlmPath)
+	item := raw.(*Item)
+
+	rawSorted, _ := exhibition.itemsSorted.Get(item.id.String())
+	itemSorted := rawSorted.(*Item)
+
+	// 0 upvotes by default
+	urequire.Equal(t, item.upvote.Size(), 0)
+
+	std.TestSetRealm(adminRealm)
+
+	urequire.NotPanics(t, func() {
+		Upvote(rlmPath)
+	})
+
+	// Check both trees for 1 upvote
+	uassert.Equal(t, item.upvote.Size(), 1)
+	uassert.Equal(t, itemSorted.upvote.Size(), 1)
+
+	// Check double upvote
+	uassert.PanicsWithMessage(t, ErrDoubleUpvote.Error(), func() {
+		Upvote(rlmPath)
+	})
+}
+
+func TestDownvote(t *testing.T) {
```

Comment:
Check this test, it is badly copy/pasted from the upvote one.

Reply:
I checked it; only comments were wrong; otherwise it should be good, no? [ff06493](https://github.com/gnolang/gno/pull/2842/commits/ff06493e12c7dae579ea2c0cbcb9d3eee276f4f7)